File: daily/cpp/Nov-24-Max-number-of-coins-you-cat-get.cpp
#include <vector>

using namespace std;
class Solution {
public:
  int maxCoins(vector<int>& piles) {
    int len = piles.size();
    int res = 0;
    sort(piles.begin(), piles.end());
    vector<int> piles_slice(piles.begin() + (len / 3), piles.end());
    for (int i = 0; i < len / 3 * 2; i+= 2) {
      res += piles_slice[i];
    }
    return res;
  }
}

File: daily/python/Nov-24-Max-number-of-coins-you-cat-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        pile_len = len(piles)
        res = 0
        piles.sort() # sort piles
        piles_slice = piles[pile_len // 3 :]
        for i in range(0, pile_len // 3 * 2, 2):
            res += piles_slice[i]

        return res
